decidim/decidim
# Fix removal of the address from proposals

#### :tophat: What? Why?
Right now, if the user wants to remove the address from the proposal, it is not possible. There is also a bug that prevents sending the form if the user first clears the address field and then unchecks the "has address" checkbox. This fixes these issues.

There are also some other usability issues with the address field geocoding that this addresses:
If you first create a proposal and the go edit it, the location point will be re-calculated in the backend (which can cause a different result) because the user did not do any selection from the geocoding autocompletion results when they submit the form. This fixes the issue as follows:
- If the record has been already geocoded, the location point will be added to the edit form
- In the front-end, if the address field is associated with a location point, it will be added to the submitted form fields to prevent re-calculation of the location point
- If the user changes the text value of the address field and exits the field, the location point will be removed causing the backend to re-calculate the result as the location has changed
- When changing the address, if the user selects one of the new geocoding results, that location point will be used even if the user later edits the address formatting (assuming they just want to refine how the address shows up on the page)
- As of #6291, it is now possible to also refine the location point in preview, so I would believe this is more closely to what the user wants

#### :pushpin: Related Issues
- Related to #6340, #6291

#### Testing
- Configure geocoding for the instance and enable geocoding for the proposals component
- Go to proposals component as a participant
- Create a new proposal with an address
- Proceed to the preview
- From preview, go back to edit proposal
- Empty the address field
- Uncheck the "has location" checkbox
- Try to submit the form (previously it wasn't possible)
- Also see in the next view that the location was actually removed (it couldn't be removed before)

#### :clipboard: Checklist
:rotating_light: Please review the [guidelines for contributing](../CONTRIBUTING.adoc) to this repository.
